packages/lib-utils/src/app/redux/ReduxExtensionProvider.test.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import type { ReactReduxContextValue } from 'react-redux';
+import { createSelectorHook } from 'react-redux';
+import type { Store } from 'redux';
+import { createStore } from 'redux';
+import ReduxExtensionProvider from './ReduxExtensionProvider';
+
+type LRReduxReducer = LoadedExtension<ResolvedExtension<ReduxProvider>>;
+
+jest.mock('@openshift/dynamic-plugin-sdk', () => ({
+  useResolvedExtensions: jest.fn(),
+}));
+
+const useResolvedExtensionsMock = jest.mocked(useResolvedExtensions, false);
+
+describe('ReduxExtensionProvider', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should register multiple providers with separate redux contexts', () => {
+    type State = {
+      value: number;
+    };
+
+    const store1: Store = createStore((state) => state ?? { value: 1 });
+    const store2: Store = createStore((state) => state ?? { value: 2 });
+
+    // create 2 core.redux-provider extensions
+    const extensions: LRReduxReducer[] = [
+      {
+        uid: '1',
+        pluginName: 'test',
+        type: 'core.redux-provider',
+        properties: {
+          store: store1,
+          context: React.createContext<ReactReduxContextValue>({
+            store: store1,
+            storeState: store1.getState(),
+          }),
+        },
+      },
+      {
+        uid: '2',
+        pluginName: 'test',
+        type: 'core.redux-provider',
+        properties: {
+          store: store2,
+          context: React.createContext<ReactReduxContextValue>({
+            store: store2,
+            storeState: store2.getState(),
+          }),
+        },
+      },
+    ];
+
+    // mock that returns the test extensions
+    useResolvedExtensionsMock.mockReturnValue([extensions, true, []]);
+
+    // create 2 contextual redux selectors; one for each extension
+    const useSelector1 = createSelectorHook<State>(extensions[0].properties.context);
+    const useSelector2 = createSelectorHook<State>(extensions[1].properties.context);
+
+    let value1 = -1;
+    let value2 = -1;
+
+    // this component will select values separately from each redux store
+    const Test: React.FC = () => {
+      value1 = useSelector1((state) => state.value);
+      value2 = useSelector2((state) => state.value);
+      return null;
+    };
+
+    render(
+      <ReduxExtensionProvider>
+        <Test />
+      </ReduxExtensionProvider>,
+    );
+
+    // assert that the values received by each selector relate to their corresponding redux state
+    expect(value1).toBe(1);
+    expect(value2).toBe(2);
+  });
+
+  test('should render children', () => {
+    // mock return value of empty set of extensions
+    useResolvedExtensionsMock.mockReturnValue([[], true, []]);
+
+    const { container } = render(
+      <ReduxExtensionProvider>
+        <span id="test" />
+      </ReduxExtensionProvider>,
+    );
+
+    expect(container.firstElementChild?.tagName).toBe('SPAN');
```

Comment:
Just wondering, does the tag name have to be uppercase?

Reply:
yes, `tagName` is always upper case